config/config.go

```diff
@@ -39,10 +39,11 @@ const defaultSamplingProbability = 0.001
 
 // Configuration configures and creates Jaeger Tracer
 type Configuration struct {
-	Disabled   bool            `yaml:"disabled"`
-	Sampler    *SamplerConfig  `yaml:"sampler"`
-	Reporter   *ReporterConfig `yaml:"reporter"`
-	RPCMetrics bool            `yaml:"rpc_metrics"`
+	Disabled            bool            `yaml:"disabled"`
+	Sampler             *SamplerConfig  `yaml:"sampler"`
+	Reporter            *ReporterConfig `yaml:"reporter"`
+	RPCMetrics          bool            `yaml:"rpc_metrics"`
+	BaggageRestrictions bool            `yaml:"baggage_restrictions"`
```

Comment:
opt-in makes sense. We can implement opt-out in the internal libs (or even not allow it).